JetNews/app/src/main/java/com/example/jetnews/ui/JetnewsApp.kt

```diff
@@ -53,22 +58,40 @@ fun JetnewsApp(
 
             val navBackStackEntry by navController.currentBackStackEntryAsState()
             val currentRoute = navBackStackEntry?.destination?.route ?: MainDestinations.HOME_ROUTE
-            Scaffold(
-                scaffoldState = scaffoldState,
-                drawerContent = {
-                    AppDrawer(
-                        currentRoute = currentRoute,
-                        navigateToHome = { navController.navigate(MainDestinations.HOME_ROUTE) },
-                        navigateToInterests = { navController.navigate(MainDestinations.INTERESTS_ROUTE) },
-                        closeDrawer = { coroutineScope.launch { scaffoldState.drawerState.close() } }
+
+            BoxWithConstraints {
+                val windowSize = getWindowSize(maxWidth)
+                Scaffold(
```

Comment:
The approach looks good to me overall, although it made me notice this `Scaffold` here, meaning there's two scaffolds in play. If we put this `DrawerContent` in each screen, I think we would lose the nice animation of destinations switching out while the drawer closes.

I also noticed that you can still open the drawer on the article detail screen, which seems a little strange.

Reply:
Nothing has changed regarding the Drawer behavior, it should work as it used to be. We just hide it when the `NavRail` needs to be shown.

Reply:
Being able to open the drawer from the article screen used to happen as well before, let's try to tackle that in a separate CL